### ISSUE TITLE
Abillity to delete old transactions.

### ISSUE THREAD

**Issue by GuardianLiarus** [bitcoin-core/gui#412]
**Is your feature request related to a problem? Please describe.**

As a Vertcoin miner, i get a lot of small payments due pools having a low minimum payout ammount, and all those transactions gets spammed to my Vertcoin Core wallet (which is based on Bitcoin Core), i have arrived to around 1500 transactions in a span of ~3 months, and now when i open my wallet, it freezes a little when loading all the transactions, it would be great if we could choose a transaction / a range of transactions and delete them from the wallet.

I also thought about maybe a toggle option that is off by default, that allows you to "keep only the transaction history from the past X days"

**Comment by katesalazar** [comment]
Hi

> [...] it would be great if we could choose a transaction / a range of transactions and delete them from the wallet.

I think wallets hold keys and transactions are in blockchains? I'm not sure what you are asking for.

While there isn't any better for you, have you thought about having two wallets? A "fast" wallet where you receive micropayments and aggregate utxos (maybe using command line interface, if you perceive a performance improvement) to send them to a "slow" wallet with fewer transactions which loads fast in the Qt client.

> I also thought about maybe a toggle option that is off by default, that allows you to "keep only the transaction history from the past X days"

Now the blockchain is growing at a more or less predictable pace, you could use "prune", meant to prune by (exact) size, in order to prune by (approximate) time.

```
$ ./src/qt/bitcoin-qt -help | grep -A10 '  -prune'
  -prune=<n>
       Reduce storage requirements by enabling pruning (deleting) of old
       blocks. This allows the pruneblockchain RPC to be called to
       delete specific blocks, and enables automatic pruning of old
       blocks if a target size in MiB is provided. This mode is
       incompatible with -txindex, -coinstatsindex and -rescan. Warning:
       Reverting this setting requires re-downloading the entire
       blockchain. (default: 0 = disable pruning blocks, 1 = allow
       manual pruning via RPC, >=550 = automatically prune block files
       to stay under the specified target size in MiB)
```

**Comment by GuardianLiarus** [comment]
> Hi
> 
> > [...] it would be great if we could choose a transaction / a range of transactions and delete them from the wallet.
> 
> I think wallets hold keys and transactions are in blockchains? I'm not sure what you are asking for.
> 
> While there isn't any better for you, have you thought about having two wallets? A "fast" wallet where you receive micropayments and aggregate utxos (maybe using command line interface, if you perceive a performance improvement) to send them to a "slow" wallet with fewer transactions which loads fast in the Qt client.
> 
> > I also thought about maybe a toggle option that is off by default, that allows you to "keep only the transaction history from the past X days"
> 
> Now the blockchain is growing at a more or less predictable pace, you could use "prune", meant to prune by (exact) size, in order to prune by (approximate) time.
> 
> ```
> $ ./src/qt/bitcoin-qt -help | grep -A10 '  -prune'
>   -prune=<n>
>        Reduce storage requirements by enabling pruning (deleting) of old
>        blocks. This allows the pruneblockchain RPC to be called to
>        delete specific blocks, and enables automatic pruning of old
>        blocks if a target size in MiB is provided. This mode is
>        incompatible with -txindex, -coinstatsindex and -rescan. Warning:
>        Reverting this setting requires re-downloading the entire
>        blockchain. (default: 0 = disable pruning blocks, 1 = allow
>        manual pruning via RPC, >=550 = automatically prune block files
>        to stay under the specified target size in MiB)
> ```

Oh yeah i know that all transactions are in the blockchain, what i am requesting is purely a visual / quality of life feature, i guess "delete" transactions wasn't the right word, it was more about hiding them / not showing them i guess, so i don't have to load as much stuff when opening my wallet.

**Comment by katesalazar** [comment]
Hi GuardianLiarus.

I think you want a more simple (at least in visual appearance) interface, with faster boot, fewer elements initially but buttons that trigger deferred data loads.

I suspect, this would be desirable, but would end up being large enough of a change, that whenever it is added to the code, you'll already fixed your problem by moving to a setup with specialized wallets.

If you ever move to a setup with specialized wallets, could you write some followup on the perceived performance difference?

**Comment by GuardianLiarus** [comment]
> Hi GuardianLiarus.
> 
> I think you want a more simple (at least in visual appearance) interface, with faster boot, fewer elements initially but buttons that trigger deferred data loads.
> 
> I suspect, this would be desirable, but would end up being large enough of a change, that whenever it is added to the code, you'll already fixed your problem by moving to a setup with specialized wallets.
> 
> If you ever move to a setup with specialized wallets, could you write some followup on the perceived performance difference?

Sorry, what do you mean by specialized wallets ?, basically having a different wallet for X purpose ?

**Comment by katesalazar** [comment]
yup

did you do that?

**Comment by GuardianLiarus** [comment]
> 

yeah, that's what i always did, it's just that the lag due to loading all those things is a bit inconvenient, but i've always been running that way yeah, i use a trezor as my main walet but vertcoin core for the one i mine to.
